xmake/rules/c++/modules/modules_support/clang.lua

```diff
@@ -149,11 +151,12 @@ function generate_stl_headerunits(target, batchcmds, headerunits, opt)
             batchcmds:vrunv(compinst:program(), table.join(compinst:compflags({target = target}), args))
         end
 
-        batchcmds:set_depmtime(os.mtime(bmifile))
-        batchcmds:set_depcache(target:dependfile(bmifile))
-
         table.insert(flags, modulefileflag .. bmifile)
+        table.append(bmifiles, bmifile)
+        depmtime = math.max(depmtime, os.mtime(bmifile))
     end
+    batchcmds:set_depmtime(depmtime)
+    batchcmds:set_depcache(target:dependfile(bmifiles))
```

Comment:
target:dependfile can only accept string

Reply:
how should i format this string ? juste append all paths ?

Reply:
or you can remove `batchcmds:set_depcache`, it will use tmpfile as depcache automatically

Reply:
```lua
function batchcmds:set_depcache(cachefile)
    local deps = self._DEPS or {}
    deps.dependfile = cachefile
    self._DEPS = deps
end
```

```lua
    -- get dependfile
    local dependfile = opt.dependfile
    if not dependfile then
        dependfile = project.tmpfile(table.concat(table.wrap(opt.files), ""))
    end
```

It will use `batchcmds:add_depfiles` to calculate a tmpfile as depcache file path.